code/__HELPERS/light_eater.dm

```diff
@@ -0,0 +1,28 @@
+// Procs that are re-used by both the light eater component and element to prevent copypasta and duplicate signals
+
+/// Abstracted as a global proc so I can avoid sending the signal from two places and reduce copypasta
+/proc/light_eater_table_buffet(atom/commisary, datum/light_eater)
```

Comment:
Could you explain a bit why this proc is not atom-typed, and why you are passing the `light_eaten` (component I assume) as an argument?

Reply:
That would require remembering what I was thinking a month ago.